pydantic/dataclasses.py

```diff
@@ -0,0 +1,43 @@
+from dataclasses import dataclass as dc
+from .main import validate_model, create_model
+
+
+def pydantic_model(cls):
+    fields = {name: (field.type, field.default)
+              for name, field
+              in cls.__dataclass_fields__.items()}
+    model = create_model('', **fields)
+    return model
+
+
+def dataclass(_cls=None, *, init=True, repr=True, eq=True,
+              order=False, unsafe_hash=False, frozen=False):
+    def wrap(cls):
+        post_init_orig = getattr(cls, '__post_init__', lambda self: None)
```

Comment:
`cls.__post_init__ = getattr(cls, '__post_init__', None)`

then below `cls.__post_init__ and cls.__post_init__(self)`

Reply:
__post_init__ [method must exist](https://github.com/ericvsmith/dataclasses/blob/d4c87c663feb0617fd8c7e6fe0a58882c4ccd468/dataclasses.py#L858) at the moment you apply the native dataclass decorator. Otherwise, it won't be called during the instance creation. In the case cls already has its own __post_init__, I save it and call it from my new __post_init__.

Reply:
ok, but I think

```py
post_init_orig = getattr(cls, '__post_init__', None)

...

def post_init(self):
    post_init_orig and post_init_orig(self)
    ...
```

Would be clearer.